lib/internal/process/warning.js

```diff
@@ -89,8 +95,18 @@ function doEmitWarning(warning) {
   process.emit('warning', warning);
 }
 
+let disableWarnings;
+
 function onWarning(warning) {
+  if (!disableWarnings) {
+    const disableWarningsValues = getOptionValue('--disable-warnings');
+    disableWarnings = new SafeSet(disableWarningsValues);
+  }
+  if ((warning?.code && SetPrototypeHas(disableWarnings, warning.code)) ||
+  (warning?.name && SetPrototypeHas(disableWarnings, warning.name))) return;
```

Comment:
`SafeSet` methods are safe to use, no need for primordials
```suggestion
  if ((warning?.code && disableWarnings.has(warning.code)) ||
      (warning?.name && disableWarnings.has(warning.name))) return;
```